### ISSUE TITLE
[BUG] ProxyOptions passed in ClientSecretCredentialBuilder.clientOptions() has no effect

### ISSUE THREAD

**Issue by turcsanyip** [Azure/azure-sdk-for-java#36296]
**Describe the bug**
I want to use `BlobServiceClient` with Service Principal authentication from behind a forward proxy.
I use `ClientSecretCredentialBuilder` to build the SP credential.
If I pass the `ProxyOptions` in `ClientSecretCredentialBuilder.clientOptions()`, it has no effect and it cannot access `login.microsoftonline.com` via my proxy.
`ManagedIdentityCredentialBuilder` is also affected.

It may be a similar issue as #28900 / #28783.

There exists a workaround: use `NettyAsyncHttpClientBuilder.proxy()` and then pass the built netty client in `ClientSecretCredentialBuilder.httpClient()`.
But the simpler API call (without building a new `HttpClient`) should work too.

***Code Snippet***
```
        ProxyOptions proxyOptions = new ProxyOptions(...);
        
        ClientSecretCredential credential = new ClientSecretCredentialBuilder()
                .tenantId(...)
                .clientId(...)
                .clientSecret(...)
                .clientOptions(new HttpClientOptions().setProxyOptions(proxyOptions))
                .build();
```

**Expected behavior**
`ProxyOptions` passed in `ClientSecretCredentialBuilder.clientOptions()` should work.

**Comment by joshfree** [comment]
@billwert @g2vinay can you please follow up on this before the `August` release?

**Comment by billwert** [comment]
@turcsanyip Hi! What version of azure-identity are you using? Taking a look.

**Comment by turcsanyip** [comment]
@billwert
azure-identity: 1.9.2
azure-storage-blob: 12.23.0

**Comment by joshfree** [comment]
https://learn.microsoft.com/en-us/azure/developer/java/sdk/proxying#use-an-explicit-proxy

**Comment by g2vinay** [comment]
@turcsanyip 

Thank you for reporting this issue.
The recommendation we have for users is to specify the proxy directly on the http client, which you've tried as a workaround.
The steps are also mentioned here: https://learn.microsoft.com/en-us/azure/developer/java/sdk/proxying#use-an-explicit-proxy
It provides better control on the client configuration for advanced scenarios like these.

**Comment by turcsanyip** [comment]
I understand that the proxy can be set on the http client along with other advanced configurations.

However, the [documentation](https://github.com/Azure/azure-sdk-for-java/blob/93a5134782e7eb70fa0114d861dc3e4919c4db59/sdk/identity/azure-identity/src/main/java/com/azure/identity/CredentialBuilderBase.java#L124-L147) of `CredentialBuilderBase.clientOptions()` also says:

> * Allows for setting common properties such as application ID, headers, proxy configuration, etc.

So in my opinion that should work too.

I haven't checked the code but my guess is that the fix may be something similar as in #28783 / [PR 29588](https://github.com/Azure/azure-sdk-for-java/pull/29588/files) (which is fairly simple, not counting the tests).

**Comment by g2vinay** [comment]
@turcsanyip 

Yeah, the client options ideally should work too.
We're investigating the root cause and a fix will be made accordingly.
the recommended way will still be to set it directly on the http client.
